sigstore/_internal/rekor/client.py

```diff
@@ -217,49 +125,37 @@ def from_response(cls, dict_: Dict[str, Any]) -> RekorLogInfo:
         )
 
 
-class RekorInclusionProof(BaseModel):
+class RekorClientError(Exception):
     """
-    Represents an inclusion proof for a Rekor log entry.
+    A generic error in the Rekor client.
     """
 
-    log_index: StrictInt = Field(..., alias="logIndex")
-    root_hash: StrictStr = Field(..., alias="rootHash")
-    tree_size: StrictInt = Field(..., alias="treeSize")
-    hashes: List[StrictStr] = Field(..., alias="hashes")
-
-    class Config:
-        allow_population_by_field_name = True
-
-    @validator("log_index")
-    def _log_index_positive(cls, v: int) -> int:
-        if v < 0:
-            raise ValueError(f"Inclusion proof has invalid log index: {v} < 0")
-        return v
-
-    @validator("tree_size")
-    def _tree_size_positive(cls, v: int) -> int:
-        if v < 0:
-            raise ValueError(f"Inclusion proof has invalid tree size: {v} < 0")
-        return v
-
-    @validator("tree_size")
-    def _log_index_within_tree_size(
-        cls, v: int, values: Dict[str, Any], **kwargs: Any
-    ) -> int:
-        if "log_index" in values and v <= values["log_index"]:
-            raise ValueError(
-                "Inclusion proof has log index greater than or equal to tree size: "
-                f"{v} <= {values['log_index']}"
-            )
-        return v
+    pass
 
 
-class RekorClientError(Exception):
+def entry_from_response(dict_: Dict[str, Any]) -> RekorEntry:
```

Comment:
I made this a helper within the internal client module rather than a class method of the `RekorEntry` since users shouldn't need this.

Reply:
Perhaps just make it a "private" class method instead? `RekorEntry._from_response` would probably be fine.

(I can look into that.)

Reply:
ceacba0..23eb302